pkg/operators/process/process.go

```diff
@@ -0,0 +1,777 @@
+	Uid           uint32    `json:"uid"`           // UID of the process owner
+	StartTime     uint64    `json:"startTime"`     // Process start time (clock ticks since system boot)
+	StartTimeStr  time.Time `json:"startTimeStr"`  // Process start time as a formatted string
+	MountNsID     uint64    `json:"mountnsid"`     // Mount namespace ID
+}
+
+type processOperator struct{}
+
+func (p *processOperator) Name() string {
+	return Name
+}
+
+func (p *processOperator) Init(globalParams *params.Params) error {
+	return nil
+}
+
+func (p *processOperator) GlobalParams() api.Params {
+	return api.Params{}
+}
+
+func (p *processOperator) InstanceParams() api.Params {
+	return api.Params{}
+}
+
+func (p *processOperator) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	// Get configuration from viper
+	config, ok := gadgetCtx.GetVar("config")
+	if !ok {
+		return nil, fmt.Errorf("config not found in gadget context")
+	}
+
+	viperConfig, ok := config.(*viper.Viper)
+	if !ok {
+		return nil, fmt.Errorf("config is not a viper instance")
+	}
+
+	// Check if process monitoring is enabled
+	enabled := viperConfig.GetBool(configKeyEnabled)
+	if !enabled {
+		gadgetCtx.Logger().Debug("Process monitoring is disabled")
+		return nil, nil
+	}
+
+	// Get the interval from config or use default
+	interval := viperConfig.GetDuration(configKeyInterval)
+	if interval <= 0 {
+		interval = defaultInterval
+		gadgetCtx.Logger().Debugf("Using default interval: %s", interval)
+	}
+
+	// Get the enabled fields from config
+	enabledFields := make(map[string]bool)
+
+	// Get fields from config
+	fields := viperConfig.GetStringSlice(configKeyFields)
+
+	// If no fields are specified, enable all fields by default
+	if len(fields) == 0 {
+		gadgetCtx.Logger().Debug("No fields specified, enabling all fields")
+		enabledFields[fieldPID] = true
+		enabledFields[fieldPPID] = true
+		enabledFields[fieldCommand] = true
+		enabledFields[fieldCPUUsage] = true
+		enabledFields[fieldMemoryRSS] = true
+		enabledFields[fieldMemoryVirtual] = true
+		enabledFields[fieldThreadCount] = true
+		enabledFields[fieldState] = true
+		enabledFields[fieldUid] = true
+		enabledFields[fieldStartTime] = true
+	} else {
+		// Enable only the specified fields
+		for _, field := range fields {
+			enabledFields[field] = true
+		}
+
+		// PID field is always required
+		enabledFields[fieldPID] = true
+
+		gadgetCtx.Logger().Debugf("Enabled fields: %v", fields)
+	}
+
+	// Create a data source for process information
+	ds, err := gadgetCtx.RegisterDataSource(datasource.TypeArray, "processes")
+	if err != nil {
+		return nil, fmt.Errorf("registering processes data source: %w", err)
+	}
+
+	instance := &processOperatorInstance{
+		interval:      interval,
+		enabledFields: enabledFields,
+		done:          make(chan struct{}),
+		dataSource:    ds,
+	}
+
+	// Add fields to the data source based on enabled fields
```

Comment:
nit: Add descriptions and annotations to all the fields.

Reply:
Added description + templates.